### PR TITLE
[BACKLOG-16728] - Using relative context path in WebContext

### PR DESCRIPTION
**Pull request by rmansoor**
- Return false to shouldUseFullyQualifiedUrl to make sure CONTEXT_PATH is relative

### CODE REVIEW

**Comment by rmansoor** [comment]
@nantunes , @pentaho-nbaker please review

File: extensions/src/main/java/org/pentaho/platform/web/http/filters/PentahoWebContextFilter.java [hunk 1]
```diff
@@ -197,16 +197,10 @@ public void doFilter( ServletRequest request, ServletResponse response, FilterCh
...
-    if ( useFullyQualifiedUrlParameter != null ) {
-      return "true".equals( useFullyQualifiedUrlParameter );
-    } else {
-      final String referer = httpRequest.getHeader( "referer" );
-      final String serverAddress = httpRequest.getScheme() + "://" + httpRequest.getServerName() + ":" + httpRequest.getServerPort();
 
-      return referer != null && !referer.startsWith( serverAddress );
-    }
+  private boolean shouldUseFullyQualifiedUrl( HttpServletRequest httpRequest ) {
+    // Will return false for now.
```
Comment (nantunes) [inline extensions/src/main/java/org/pentaho/platform/web/http/filters/PentahoWebContextFilter.java]: If not removing it completely, I would keep the code for the explicit use of the `fullyQualifiedUrl` parameter and just always return `false` inside the `else` clause (so it doesn't try to be _"smart"_).

Also include the BACKLOG-16728 reference in the comment, so in the future is it easier to find out why it is returning false for now.

File: extensions/src/test/java/org/pentaho/platform/web/http/filters/PentahoWebContextFilterTest.java [hunk 2]
```diff
@@ -162,54 +162,6 @@ public void testDoGetWithNoOsgiRequireConfigDefault() throws ServletException, I
     assertTrue( this.requirejsManagerInitIsCalled( response, null ) );
   }
 
-  @Test
-  public void testDoGetWithFullyQualifiedUrlTrue() throws ServletException, IOException {
```
Comment (nantunes) [inline extensions/src/test/java/org/pentaho/platform/web/http/filters/PentahoWebContextFilterTest.java]: Recover the first two tests, that deal with the `fullyQualifiedUrl` parameter.

You can also modify the other two to tests to verify it uses the relative path by default.

**Comment by dcleao** [comment]
The `fullyQualifiedUrl` parameter is an important feature that enables **embeddability** of whatever we do in the web/JavaScript side of the Pentaho platform. This includes the Visualization API. CDF can also start using this new capability to be able to drop the custom `cdf-embed.js`/`cde-embed.js` endpoints it had to develop in the past.

**Comment by nantunes** [comment]
@dcleao right. It might, probably, fail the same way it is failing on the proxy scenario (although it did work in our sandboxed embedded scenario).
We need a story for figuring out what plugins aren't handling the full URL correctly.

But keeping the parameter is harmless, so only the change that modified existing default behaviour must be reverted.

**Comment by dcleao** [comment]
@nantunes Good point.

**Comment by rmansoor** [comment]
@nantunes . I have updated the PR with the changes you have suggested. I have  tested with the changes and all is working well. Please review the updated PR now

**Pull request merged**